src/lib/picker/category.component.ts

```diff
@@ -1,80 +1,110 @@
-      </span>
-    </div>
-
-    <ng-template [ngIf]="emojis">
-      <ngx-emoji
-        *ngFor="let emoji of emojis; trackBy: trackById"
-        [emoji]="emoji"
-        [size]="emojiSize"
-        [skin]="emojiSkin"
-        [isNative]="emojiIsNative"
-        [set]="emojiSet"
-        [sheetSize]="emojiSheetSize"
-        [forceSize]="emojiForceSize"
-        [tooltip]="emojiTooltip"
-        [backgroundImageFn]="emojiBackgroundImageFn"
-        [hideObsolete]="hideObsolete"
-        (emojiOver)="emojiOver.emit($event)"
-        (emojiLeave)="emojiLeave.emit($event)"
-        (emojiClick)="emojiClick.emit($event)"
-      ></ngx-emoji>
-    </ng-template>
+    <section
+      #container
+      class="emoji-mart-category"
+      [attr.aria-label]="i18n.categories[id]"
+      [class.emoji-mart-no-results]="emojis && !emojis.length"
+      [ngStyle]="containerStyles"
+    >
+      <div class="emoji-mart-category-label" [ngStyle]="labelStyles" [attr.data-name]="name">
+        <!-- already labeled by the section aria-label -->
+        <span #label [ngStyle]="labelSpanStyles" aria-hidden="true">
+          {{ i18n.categories[id] }}
+        </span>
+      </div>
 
-    <div *ngIf="emojis && !emojis.length">
-      <div>
+      <div
+        *ngIf="virtualize && filteredEmojis$ | async as filteredEmojis; else normalRenderTemplate"
+      >
         <ngx-emoji
-          [emoji]="notFoundEmoji"
-          size="38"
+          *ngFor="let emoji of filteredEmojis; trackBy: trackById"
+          [emoji]="emoji"
+          [size]="emojiSize"
           [skin]="emojiSkin"
           [isNative]="emojiIsNative"
           [set]="emojiSet"
           [sheetSize]="emojiSheetSize"
           [forceSize]="emojiForceSize"
           [tooltip]="emojiTooltip"
           [backgroundImageFn]="emojiBackgroundImageFn"
+          [imageUrlFn]="emojiImageUrlFn"
+          [hideObsolete]="hideObsolete"
           [useButton]="emojiUseButton"
+          (emojiOver)="emojiOver.emit($event)"
+          (emojiLeave)="emojiLeave.emit($event)"
+          (emojiClick)="emojiClick.emit($event)"
         ></ngx-emoji>
       </div>
 
-      <div class="emoji-mart-no-results-label">
-        {{ i18n.notfound }}
+      <div *ngIf="emojis && !emojis.length">
+        <div>
+          <ngx-emoji
+            [emoji]="notFoundEmoji"
+            size="38"
+            [skin]="emojiSkin"
+            [isNative]="emojiIsNative"
+            [set]="emojiSet"
+            [sheetSize]="emojiSheetSize"
+            [forceSize]="emojiForceSize"
+            [tooltip]="emojiTooltip"
+            [backgroundImageFn]="emojiBackgroundImageFn"
+            [useButton]="emojiUseButton"
+          ></ngx-emoji>
+        </div>
+
+        <div class="emoji-mart-no-results-label">
+          {{ i18n.notfound }}
+        </div>
       </div>
-    </div>
+    </section>
 
-  </section>
+    <ng-template #normalRenderTemplate>
+      <div *ngIf="!virtualize && emojis">
```

Comment:
is `!virtualize` needed here for the else template?

Reply:
I did this because filteredEmojis$ can be null while emojis can have a value. When virtualized is true, it can render the normal template in this case. I'll add an ng-template for the filteredEmojis$ check above so that we are not handling virtualized cases in the "normalRenderTemplate".